### PR TITLE
GEN-1569 - refact: retrive product reviews information from ProductData context

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Update `ProductData` type so it includes `averageRating` and `reviewComments` info;
* Update `fetchProductData` function so it fetches `averageRating` and `reviewComments`;
* Update codebase by getting `averageRating` and `reviewComments` from `useProductData()` instead of `useProductPageContext()`

## Justify why they are needed

It makes more sense that this data comes from `ProductData` Context given its nature.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3805?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3804** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3804?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3805** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3805?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #3806** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3806?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3805?utm_source=stack-comment).

File: apps/store/src/components/ProductData/fetchProductData.ts [hunk 1]
```diff
@@ -23,5 +27,12 @@ export const fetchProductData = async ({
     throw new Error(`Unable to fetch product data: ${variables.productName}`)
   }
 
-  return data.product
+  const averageRating = await getProductAverageRating(variables.productName)
+  const reviewComments = await getProductReviewComments(variables.productName)
```
Comment (robinandeer) [inline apps/store/src/components/ProductData/fetchProductData.ts:31]: should we not run this in parallel?

**Review by robinandeer — COMMENTED** [review]
Product data is potentially one of those awkward cases where we fetch data from the API and instead of using Apollo Cache, we create a new manual context to access it...

Have you considered creating a separate context only for product review data?

File: apps/store/src/components/ProductData/fetchProductData.ts (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/ProductData/fetchProductData.ts:31]: Yes, we should 👍🏻

**Pull request closed without merging**